hs_tools_resource/app_launch_helper.py

```diff
@@ -9,39 +9,54 @@ def resource_level_tool_urls(resource_obj, request_obj):
     res_type_str = resource_obj.resource_type
 
     tool_list = []
+    tool_res_id_list = []
     open_with_app_counter = 0
 
+    # associate resources with app tools using extended metadata name-value pair with 'appkey' key
+    tool_app_key = 'appkey'
+    tool_kv_dict = {}
+    for tool_res_obj in ToolResource.objects.all():
```

Comment:
@hyi I think you can simplify this code using the hstore field query api. Have a look at the documentation: https://docs.djangoproject.com/en/1.8/ref/contrib/postgres/fields/#querying-hstorefield

The code needed to find a list of web apps associated with a given resource can be done with only couple of lines of code once you use the hstore specific query api.

Reply:
@pkdash Thanks for the info. Will update the code to use the hstore field query api.